dogpile/cache/region.py

```diff
@@ -1,23 +1,32 @@
 from __future__ import with_statement
-from .. import Lock, NeedRegenerationException
-from ..util import NameRegistry
-from . import exception
-from ..util import PluginLoader, memoized_property, coerce_string_conf
-from .util import function_key_generator, function_multi_key_generator
-from .api import NO_VALUE, CachedValue
-from .proxy import ProxyBackend
-from ..util import compat
-import time
+
 import datetime
+from functools import partial
+from functools import wraps
 from numbers import Number
-from functools import wraps, partial
 import threading
+import time
+
 from decorator import decorate
 
+from . import exception
+from .api import CachedValue
+from .api import NO_VALUE
+from .proxy import ProxyBackend
+from .util import function_key_generator
+from .util import function_multi_key_generator
+from .. import Lock
+from .. import NeedRegenerationException
+from ..util import coerce_string_conf
+from ..util import compat
+from ..util import memoized_property
+from ..util import NameRegistry
+from ..util import PluginLoader
+
 _backend_loader = PluginLoader("dogpile.cache")
 register_backend = _backend_loader.register
-from . import backends  # noqa
 
+from . import backends  # noqa
```

Comment:
make the comment here "# noqa nosort".    part of the black/zimports thing is you can safely run black / zimports on the source file and it should not change.

Reply:
doesn't work... I've try with:
```python
from . import backends  # noqa nosort
```

The problem come from the lines bellow who are moved after the import:
```python
_backend_loader = PluginLoader("dogpile.cache")
register_backend = _backend_loader.register
```
Also, I've try to use something like this but he doesn't work too:
```python
_backend_loader = PluginLoader("dogpile.cache")  # nosort
register_backend = _backend_loader.register  # nosort

from . import backends  # noqa nosort
```

`# nosort` seem to be ignored for line who are not imports... maybe a `zimports` bug/improvement.

Thoughts?

Reply:
zimports is curently looking for both at the same time, "# noqa nosort".   if you add that to all the imports that need to be in exact ordering it should leave them all alone, except that they will be after all the sorted imports.  otherwise we need to fix zimports.

Reply:
oh wait, the PluginLoader / _backend_loader arent imports, so no, that won't work.

If we use __import__() for now for that particiular import then it should be OK, or if we put it inside a block of some kind like a def.

Reply:
let me just fix dogpile here that pattern is terrible

Reply:
lets get https://gerrit.sqlalchemy.org/#/c/sqlalchemy/dogpile.cache/+/1148/ merged then do this

Reply:
ack